Lib/fontTools/subset/__init__.py

```diff
@@ -2556,8 +2556,8 @@ def _closure_glyphs(self, font):
 
 		self.glyphs_emptied = frozenset()
 		if self.options.retain_gids:
-			self.glyphs_emptied = realGlyphs - self.glyphs_retained
-			# TODO Drop empty glyphs at the end of GlyphOrder vector.
+			max_retained_glyph_id = max(font.getGlyphID(g) for g in self.glyphs_retained)
+			self.glyphs_emptied = {g for g in realGlyphs - self.glyphs_retained if font.getGlyphID(g) <= max_retained_glyph_id}
```

Comment:
There's a better way to write this one to avoid calling getGlyphID on all... Just take the first `max_raintained_glyph_id+1` members of the GlyphOrder and turn into set and subtract glyphs_retained... Feel free to ignore.

Reply:
Rewrote using last_retained_order.